### PR TITLE
Tailored Flows: Update Newsletter flow 

### PR DESCRIPTION
**Pull request by escapemanuele**
#### Proposed Changes

Fix the newsletter flow following the latest changes in the designs

#### Testing Instructions

1. Pull branch and `yarn start`
2. Start at the beginning of the flow http://calypso.localhost:3000/setup?flow=newsletter
 * test with logged in and logged out
 * test purchase and free plan
3. Check the other /start flow to make sure there was no regression

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/processing-step/index.tsx [hunk 1]
```diff
@@ -42,7 +42,7 @@ const ProcessingStep: Step = function ( props ): ReactElement | null {
 			if ( typeof action === 'function' ) {
 				try {
 					await action();
-					submit?.( {}, ProcessingResult.SUCCESS );
+					submit?.( {}, ProcessingResult.SUCCESS, props.data?.navigateTo );
```
Comment (escapemanuele) [inline client/landing/stepper/declarative-flow/internals/steps-repository/processing-step/index.tsx]: Better make sure this doesn't cause problems with regression

File: client/landing/stepper/declarative-flow/internals/steps-repository/newsletter-setup/index.tsx [hunk 2]
```diff
@@ -52,6 +52,10 @@ const NewsletterSetup: Step = ( { navigation } ) => {
 	const [ accentColor, setAccentColor ] = React.useState< string | undefined >();
 	const [ base64Image, setBase64Image ] = React.useState< string | null >();
 	const [ selectedFile, setSelectedFile ] = React.useState< File | undefined >();
+	const siteTitleError =
+		formTouched && ! siteTitle.trim()
+			? 'Your publication needs a name so your subscribers can identify you.'
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/newsletter-setup/index.tsx]: Out of scope but this is untranslated.

File: client/landing/stepper/declarative-flow/internals/steps-repository/processing-step/index.tsx (hunk 1 above)
Comment (alshakero) [reply client/landing/stepper/declarative-flow/internals/steps-repository/processing-step/index.tsx]: Do we need this? The pending action should do the work itself.

File: client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx [hunk 3]
```diff
@@ -81,17 +81,25 @@ const CompletingPurchase: Step = function CompletingPurchase( { navigation, flow
...
 		} else {
+			navigateTo = 'subscribers';
+
 			setPendingAction( async () => {
 				setProgressTitle( __( 'Completing Purchase' ) );
 				setProgress( 0.3 );
-				await wait( 2000 );
+				await wait( 1000 );
 				setProgress( 1 );
+				await wait( 2000 );
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx]: Shouldn't we just call `submit`/`navigate` here and be happy without passing `navigateTo` around?

File: client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx [hunk 4]
```diff
@@ -81,17 +81,25 @@ const CompletingPurchase: Step = function CompletingPurchase( { navigation, flow
...
 			} );
 		}
-		submit?.();
+
+		const providedDependencies = { navigateTo };
+		submit?.( providedDependencies, navigateTo );
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx]: +1

File: client/landing/stepper/declarative-flow/internals/steps-repository/subscribers/index.tsx [hunk 5]
```diff
@@ -1,9 +1,34 @@
...
+		setPendingAction( async () => {
+			setProgressTitle( __( 'Creating your Newsletter' ) );
+			setProgress( 0.3 );
+			await wait( 1500 );
+			setProgress( 1 );
+			setProgressTitle( __( 'Preparing Next Steps' ) );
+			await wait( 2000 );
+		} );
+
+		submit?.( providedDependencies, navigateTo );
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/subscribers/index.tsx]: +1. 

The pending action should call `submit`.

File: client/landing/stepper/declarative-flow/newsletter.ts [hunk 6]
```diff
@@ -54,12 +47,24 @@ export const newsletter: Flow = {
 				case 'newsletterSetup':
 					return window.location.replace( '/start/newsletter/domains' );
 
+				case 'completingPurchase':
+					return navigate( 'processing', { navigateTo: 'subscribers' } );
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/newsletter.ts]: +1.

File: client/signup/tailored-flow-processing-screen/style.scss [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+	&.is-newsletter {
+		.reskinned-processing-screen__container {
+			background-color: rgb( 162, 218, 254 );
+			background-image: url( 'calypso/assets/images/onboarding/newsletter-processing-left.png' ),
+			url( 'calypso/assets/images/onboarding/newsletter-processing-right.jpg' );
+			background-position: left bottom, right top;
+			background-repeat: no-repeat, no-repeat;
+			background-attachment: fixed;
+
+			background-size: auto 25%, auto 25%;
```
Comment (alshakero) [inline client/signup/tailored-flow-processing-screen/style.scss:67]: So neat!

**Review by alshakero — CHANGES_REQUESTED** [review]
Beautiful! Just concerned about the submit call. LMK if I'm wrong, I'm not confident about my idea.

**Comment by alshakero** [comment]
I think we should add `subsribers` step [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/internals/global.scss) to stylize it for nicer demos.

File: client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx (hunk 3 above)
Comment (escapemanuele) [reply client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx]: The problem here is:
```js
case 'completingPurchase':
	return navigate( 'processing', { navigateTo: 'subscribers' } );
case 'subscribers':
	return navigate( 'processing', { navigateTo: 'launchpad' } );
```

We have two different steps that need to show some processing. `navigateTo` is used to determine where to navigate after processing is done!
Tell me if this makes sense for you. For sure there may be a better way for this, but looked like a quick solution for better managing after processing redirection.
```js
case 'processing': {
	return navigate( params[ 1 ] as StepPath );
}
```

Comment (alshakero) [reply client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx]: How about we make the pending action return the destination?

In this [line](https://github.com/Automattic/wp-calypso/blob/update/newsletter-flow-pre-checkout/client/landing/stepper/declarative-flow/internals/steps-repository/processing-step/index.tsx#L44), we can do
```js
const destination = await action();
submit?.( { destination }, ProcessingResult.SUCCESS );
```

This way, [the pending action](https://github.com/Automattic/wp-calypso/blob/update/newsletter-flow-pre-checkout/client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx#L97) can return the destination.

Comment (escapemanuele) [reply client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx]: I really like that!

File: client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx [hunk 8]
```diff
@@ -39,60 +40,84 @@ const CompletingPurchase: Step = function CompletingPurchase( { navigation, flow
...
+			setProgressTitle( __( 'Preparing Next Steps' ) );
+
+			await postSiteLogo();
+			setProgress( 1 );
+			return 'launchpad';
+		} );
+	};
+
+	const completeNewsletterFlow = () => {
+		setPendingAction( async () => {
```
Comment (renancarvalho) [inline client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx:95]: This is getting a good shape =) 
I believe newsletter has the `accent color` to be posted too, It is available in the store as `siteAccentColor`

File: client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx [hunk 9]
```diff
@@ -39,60 +40,84 @@ const CompletingPurchase: Step = function CompletingPurchase( { navigation, flow
...
+		} );
+
+		recordTracksEvent( 'calypso_signup_site_options_submit', {
+			has_site_title: !! state.siteTitle,
+			has_tagline: !! state.siteDescription,
+		} );
+	}, [ site, saveSiteSettings, state ] );
+
+	const postSiteLogo = useCallback( async () => {
+		if ( ! state.siteLogo ) {
```
Comment (renancarvalho) [inline client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx:63]: I am not sure if it is a mandatory field, If so we need to find a way to communicate to the user, as we don't have 'goBack' buttons.. I think we could validate it on the setup step.

Perhaps this is out of scope for this PR..

File: client/landing/stepper/declarative-flow/internals/steps-repository/intro/intro.tsx [hunk 10]
```diff
@@ -42,7 +42,7 @@ const Intro: React.FC< Props > = ( { goNext, flowName } ) => {
 			<h1 className="intro__title">
 				<span>{ introContent[ flowName ].title }</span>
 			</h1>
-			<Button className="intro__button" primary onClick={ goNext }>
+			<Button className="intro__button" primary onClick={ onSubmit }>
```
Comment (renancarvalho) [inline client/landing/stepper/declarative-flow/internals/steps-repository/intro/intro.tsx:45]: Is there a specific reason to 'navigate' over the flows that are not gathering data via submit instead of 'goNext'?
Not questioning the implementation, it is just for my better understanding =)

**Review by renancarvalho — COMMENTED** [review]
These are good changes, we are getting closer to the full flow <3
I have noticed we have many `await()` on the process step to give the user a better feeling. Is it going to be final ?

File: client/landing/stepper/declarative-flow/internals/steps-repository/intro/intro.tsx (hunk 10 above)
Comment (escapemanuele) [reply client/landing/stepper/declarative-flow/internals/steps-repository/intro/intro.tsx:45]: I know that it is a bit confusing, when working on Stepper I introduced `submit` for signaling step completion, when you fulfill what's inside the step, while `goNext` was used in steps where you could skip what's inside.

File: client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx (hunk 8 above)
Comment (escapemanuele) [reply client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx:95]: You're correct, it's not saved now! 
I'm thinking where should it be saved, maybe in site settings.

**Comment by escapemanuele** [comment]
@renancarvalho thanks for your feedback. 
`wait` are totally arbitrary, added by what I felt best, but are totally modifiable!

File: client/landing/stepper/declarative-flow/internals/steps-repository/processing-step/index.tsx (hunk 1 above)
Comment (escapemanuele) [reply client/landing/stepper/declarative-flow/internals/steps-repository/processing-step/index.tsx]: Not a problem anymore as we followed another road 🎸

File: client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx (hunk 8 above)
Comment (escapemanuele) [reply client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx:95]: Saved in site settings ✌️

Comment (mashikag) [reply client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx:95]: @escapemanuele saving of the accent color overlaps with the @Automattic/apex 's project: pdDOJh-Ee-p2.

Our initial thought was to store it directly as a theme's primary color. I am not sure how this can be done just yet but it seems to be doable since it can be done by editing global styles in FSE. I am looking into this right now.

**Review by alshakero — COMMENTED** [review]
It seems the `reskinned-processing-screen` has unnecessary scroll bars.

File: client/landing/stepper/declarative-flow/internals/steps-repository/newsletter-setup/index.tsx [hunk 11]
```diff
@@ -84,13 +88,14 @@ const NewsletterSetup: Step = ( { navigation } ) => {
...
-				// await setSiteLogo( new File( [ base64ImageToBlob( base64Image ) ], 'site-logo.png' ) );
 			} catch ( _error ) {
 				// communicate the error to the user
 			}
 		}
-		// submit?.( { siteTitle, tagline } );
+
+		// if ( siteTitle.trim().length ) {
+		submit?.( { siteTitle, tagline } );
+		// }
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/newsletter-setup/index.tsx:98]: Could we bring this back? It was removed mistakenly.

File: client/landing/stepper/declarative-flow/internals/steps-repository/processing-step/index.tsx [hunk 12]
```diff
@@ -41,8 +41,8 @@ const ProcessingStep: Step = function ( props ): ReactElement | null {
 		( async () => {
 			if ( typeof action === 'function' ) {
 				try {
-					await action();
-					submit?.( {}, ProcessingResult.SUCCESS );
+					const destination = await action();
+					submit?.( { destination }, ProcessingResult.SUCCESS );
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/processing-step/index.tsx]: ✨

File: client/landing/stepper/declarative-flow/internals/steps-repository/subscribers/index.tsx [hunk 13]
```diff
@@ -1,9 +1,32 @@
...
+
+	const handleSubmit = () => {
+		setPendingAction( async () => {
+			setProgressTitle( __( 'Creating your Newsletter' ) );
+			setProgress( 0.3 );
+			await wait( 1500 );
+			setProgress( 1 );
+			setProgressTitle( __( 'Preparing Next Steps' ) );
+			await wait( 2000 );
+			return 'launchpad';
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/subscribers/index.tsx]: It may be worth it here to return 
```suggestion
			return { destination: 'launchpad' }
```

As a way to document what the return means and to allow future extensibility.

File: client/signup/config/flows-pure.js [hunk 14]
```diff
@@ -114,10 +114,13 @@ export function generateFlows( {
 		{
 			name: 'newsletter',
 			steps: [ 'domains', 'plans' ],
-			destination: ( dependencies ) =>
-				`/setup/subscribers?flow=newsletter&siteSlug=${ dependencies.siteSlug }`,
+			destination: ( dependencies, localeSlug, goesThroughCheckout ) => {
+				return goesThroughCheckout
+					? `/setup/completingPurchase?flow=newsletter&siteSlug=${ dependencies.siteSlug }`
+					: `/setup/subscribers?flow=newsletter&siteSlug=${ dependencies.siteSlug }`;
```
Comment (alshakero) [inline client/signup/config/flows-pure.js]: I think `siteSlug` can be user input, we should `encodeURIComponent`.

File: client/signup/reskinned-processing-screen/index.jsx [hunk 15]
```diff
@@ -26,6 +26,14 @@ const useSteps = ( { flowName, hasPaidDomain, isDestinationSetupSiteFlow } ) =>
 		case 'site-content-collection':
 			steps = [ { title: __( 'Saving your content' ) }, { title: __( 'Closing the loop' ) } ];
 			break;
+		case 'newsletter':
+			steps = [
+				{ title: __( 'Saving your preferences' ) },
+				{ title: __( 'Getting your Domain' ) },
+				{ title: __( 'Adding your Plan' ) },
+				{ title: __( 'Preparing Checkout' ) },
```
Comment (alshakero) [inline client/signup/reskinned-processing-screen/index.jsx:34]: The capitalization here is inconsistent. `preferences` vs `Domain` and `your` is lowercased. I think only the first character should be capitalized.

**Review by alshakero — COMMENTED** [review]
Left a few cosmetic optional comments.

File: client/signup/reskinned-processing-screen/index.jsx (hunk 15 above)
Comment (escapemanuele) [reply client/signup/reskinned-processing-screen/index.jsx:34]: I think that too, but here we just followed the designs. We should ask @saygunnyc

**Review by arcangelini — APPROVED** [review]
Looks good to me. One thing I noticed is that the color picker is missing the notice in the event the color is not going to be a visible or contrast sucks. Not sure if there was a conversation about this already but I figured I would just point it out in case.

File: client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx (hunk 8 above)
Comment (escapemanuele) [reply client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx:95]: Thanks @mashikag, we'll remove it from our PR :)

**Comment by mashikag** [comment]
> Looks good to me. One thing I noticed is that the color picker is missing the notice in the event the color is not going to be a visible or contrast sucks. Not sure if there was a conversation about this already but I figured I would just point it out in case.

Indeed there was @arcangelini: pdDOJh-G0-p2. 🙂

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7479189

Hi @escapemanuele, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Saving your preferences`
- `Getting your Domain`
- `Preparing Checkout`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7479189).
